components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/LogoutRequestProcessor.java

```diff
@@ -103,16 +103,25 @@ public SAMLSSOReqValidationResponseDTO process(LogoutRequest logoutRequest, Stri
 
                 issuer = logoutRequest.getIssuer().getValue();
 
-                String[] splitIssuer = issuer.split("@");
-                if (splitIssuer.length == 2 && !splitIssuer[0].trim().isEmpty() && !splitIssuer[1].trim().isEmpty()) {
-                    issuer = splitIssuer[0];
+                if (issuer.contains("@")) {
+                    String[] splitIssuer = issuer.split("@");
```

Comment:
Check whether the issuer name can have '@' within itself. If so need to split to 2 at the last '@'